jmix-flowui/flowui/src/main/java/io/jmix/flowui/app/bulk/BulkEditView.java

```diff
@@ -0,0 +1,586 @@
+
+        AbstractField<?, ?> editField = createField(managedField);
+        dataFields.put(managedField.getFqn(), editField);
+        container.add(editField);
+
+        JmixButton clearButton = createClearButton(editField, isFieldRequired(managedField));
+        container.add(clearButton);
+
+        return container;
+    }
+
+    protected AbstractField<?, ?> createField(ManagedField managedField) {
+        MetaProperty metaProperty = managedField.getMetaProperty();
+        ComponentGenerationContext generationContext =
+                new ComponentGenerationContext(metaProperty.getDomain(), metaProperty.getName());
+        generationContext.setTargetClass(getClass());
+        AbstractField<?, ?> field = (AbstractField<?, ?>) uiComponentsGenerator.generate(generationContext);
+
+        if (field instanceof SupportsValidation<?> supportsValidation) {
+            BulkEditBeanPropertyValidator beanValidator = getBeanPropertyValidator(metaProperty);
+            if (beanValidator != null) {
+                supportsValidation.addValidator(beanValidator);
+            }
+            //noinspection rawtypes
+            Validator customValidator = context.getFieldValidators().get(managedField.getFqn());
+            if (customValidator != null) {
+                //noinspection unchecked
+                supportsValidation.addValidator(customValidator);
+            }
+        }
+        if (field instanceof HasSize hasSize) {
+            hasSize.setMinWidth(FIELD_MIN_WIDTH);
+        }
+        return field;
+    }
+
+    @Nullable
+    protected BulkEditBeanPropertyValidator getBeanPropertyValidator(MetaProperty metaProperty) {
+        MetaClass propertyEnclosingMetaClass = metaProperty.getDomain();
+        Class<?> enclosingJavaClass = propertyEnclosingMetaClass.getJavaClass();
+
+        BeanDescriptor beanDescriptor = validator.getConstraintsForClass(enclosingJavaClass);
+        if (beanDescriptor.isBeanConstrained()) {
+            return beanValidatorProvider.getObject(enclosingJavaClass, metaProperty.getName());
+        } else {
+            return null;
+        }
+    }
+
+    protected boolean isFieldRequired(ManagedField managedField) {
+        MetaProperty metaProperty = managedField.getMetaProperty();
+        boolean requiredFromMetaProperty = metaProperty.isMandatory();
+
+        Object notNullUiComponent = metaProperty.getAnnotations()
+                .get(NotNull.class.getName() + "_notnull_ui_component");
+        boolean requiredFromAnnotations = Boolean.TRUE.equals(notNullUiComponent);
+
+        return requiredFromMetaProperty || requiredFromAnnotations;
+    }
+
+    protected JmixButton createClearButton(AbstractField<?, ?> field, boolean isFieldRequired) {
+        JmixButton button = uiComponents.create(JmixButton.class);
+        button.setIcon(VaadinIcon.TRASH.create());
+
+        if (isFieldRequired) {
+            button.getElement().getStyle().set("visibility", "hidden");
```

Comment:
`button.setVisible(false)` ?

Reply:
setVisible(false) hides button without any spacer - relative field is stretched to full width:
![image](https://github.com/jmix-framework/jmix/assets/142792860/23accd02-894d-4740-9533-491d5f3e6327)

Reply:
OK